pkg/prometheus/resource_selector.go

```diff
@@ -841,3 +846,29 @@ func (rs *ResourceSelector) validateEC2SDConfigs(ctx context.Context, sc *monito
 	}
 	return nil
 }
+
+func (rs *ResourceSelector) validateAzureSDConfigs(ctx context.Context, sc *monitoringv1alpha1.ScrapeConfig) error {
+	for i, config := range sc.Spec.AzureSDConfigs {
+		// Since Prometheus uses default authentication method as "OAuth"
+		if config.AuthenticationMethod == nil || *config.AuthenticationMethod == "OAuth" {
```

Comment:
(nit) it might be easier to continue if the authentication method is managed identity
```suggestion
		if ptr.Deref(config.AuthenticationMethod, "") == "ManagedIdentify" {
			continue
		}
```